packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -1788,6 +1788,49 @@ describe('TransactionController', () => {
 
       expect(delayMessengerMock.call).not.toHaveBeenCalled();
     });
+
+    it('catches error no code property in error object', async () => {
+      const consoleSpy = jest.spyOn(console, 'error').mockImplementation();
+
+      (
+        delayMessengerMock.call as jest.MockedFunction<any>
+      ).mockImplementationOnce(() => {
+        // eslint-disable-next-line @typescript-eslint/no-throw-literal
+        throw undefined;
+      });
+
+      const controller = newController();
+      controller.state.transactions.push(txMeta as any);
+
+      controller.initApprovals();
+      await wait(100);
+
+      expect(consoleSpy).toHaveBeenCalledTimes(1);
+      expect(consoleSpy).toHaveBeenCalledWith(
+        'Error during persisted transaction approval',
+        TypeError(`Cannot read properties of undefined (reading 'code')`),
```

Comment:
This suggests we have the same issue further down the stack, likely in the `catch` inside `processApproval` where we also check `error.code`.

I think we need the same optional chaining there so we can get a meaningful error here rather than an error during error capture.

Reply:
Great spot, I added in the `processApproval` the optional chaining. I had to modify a bit the unit test, mocking an object with 'message' cause otherwise transaction fails with `Internal JSON-RPC error` as the message property is required.